client/js/jsonedit.js

```diff
@@ -830,35 +897,89 @@ function displayComputeOps() {
 }
 
 function jeAddCSScommands() {
-  const presets = {
-    '[a-z]+': [
-      'border: 1px solid black', 'background: white', 'font-size: 30px', 'color: black' , 'border-radius: 100%'
-    ],
-    'button': [
-      '--wcMain: #1f5ca6', '--wcMainOH: #0d2f5e', '--wcBorder: #0d2f5e', '--wcBorderOH: #1f5ca6', '--wcFont: #ffffff', '--wcFontOH: #ffffff'
-    ],
-    'seat': [
-      '--wcShadowTurn: 0px 0px 20px 5px var(--color)'
-    ],
-    'timer': [
-      '--wcBorderNormal: #00000000', '--wcBorderAlert: red', '--wcFontAlert: red', '--wcFontPaused: #6d6d6d', '--wcAnimationAlert: blinker 1s linear infinite', '--wcAnimationPaused: none'
-    ]
+  const string_presets = {
+    "border": "1px solid black", "background": "white", "font-size": "16px", "color": "black", "background-image": "url('')"
+  };
+  const nested_presets = {
+    '[a-z]+': {
+      'default': string_presets,
+      ':hover': string_presets
+    },
+    'seat': {
+      '.seated.turn': {}
+    },
+    'timer': {
+      '.alert': {}, '.paused':{}
+    },
+    'holder': {
+      '.droptarget': {"border": "calc(1px / var(--scale)) solid #333 !important"}
+    },
+    'pile': {
+      '.pile .handle': {}
+    }
   };
-  for(const type in presets) {
-    for(const css of presets[type]) {
+
+  // Add nested object button items
+  for(const type in nested_presets) {
+    for(const cssSection of Object.keys(nested_presets[type])) { // Add CSS sections
       jeCommands.push({
-        id: 'css_' + css,
-        name: css,
-        context: `^${type} ↦ (css|[a-z]+CSS)`,
-        call: async function() {
-          jePasteText(css + '; ', true);
+        id: 'css_' + cssSection,
+        name: cssSection,
+        context: `^${type} ↦ css`,
+        show:  function() {
+          if(typeof jeStateNow.css != "object" || JSON.stringify(jeStateNow.css) == '{}')
+            return false;
+          for(const property in jeStateNow.css)
+            if(typeof jeStateNow.css[property] != "object")
+              return false;
+          return jeStateNow.css[cssSection] == undefined ? true : false;
         },
-        show: function() {
-          return !String(jeGetValue()[jeGetLastKey()]).match(css.split(':')[0]);
+        call: async function() {
+          jeStateNow.css[cssSection] = '###SELECT ME###';
+          jeSetAndSelect({});
         }
       });
+      for(const cssProperty of Object.keys(nested_presets[type][cssSection])) { // Add entries per-section
```

Comment:
Same question about `for(... in ...)`.

And since you use the key _and_ the value in your code, you could also consider using `for(const [ key, value ] of Object.entries(object))`.